helm/ingress-azure/templates/deployment.yaml

```diff
@@ -32,8 +32,10 @@ spec:
         {{- end }}
     spec:
       serviceAccountName: {{ template "application-gateway-kubernetes-ingress.serviceaccountname" . }}
-      securityContext:        
-        runAsUser: 0
+      {{- with .Values.securityContext }}
```

Comment:
cloud-provider config is not just used for credentials but also vnet/route table information. There could be users who are relying on cloud-provider and using service principal at the same time.

Can we just add this section for `securityContext` and revert the remaining changes related to volume mount ? Volume changes should just led to mounting of an empty volume if the pod doesn't have enough permissions.